filecache.go

```diff
@@ -16,13 +16,19 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// File is an abstraction to allow specifying an access token for the given file
+type File struct {
+	Path        string
+	AccessToken string
```

Comment:
Rather than making these so specific, I think I'd make an args map so that you don't have to break this all again in the future for another service. Something like `map[string]string`. Then it's `myFileRec.Args["AccessToken"] = "myToken"`

Reply:
Great idea, thanks!